Ch5 issue18

Syakyr opened the pull request.

jannah-aisg committed:
Initial draft

snjannah committed:
Initial draft

kwanchettan committed:
added the new documents

kwanchettan committed:
added references and renamed the titles

kwanchettan committed:
edited the subheader

kwanchettan committed:
added a draft

kwanchettan committed:
removed unwanted file

kwanchettan committed:
added diagram and chart to illustrate

kwanchettan committed:
included minor edits on the language

kwanchettan committed:
edited the language

kwanchettan committed:
amended the language

kwanchettan committed:
amended the language

chuawjk committed:
Merge branch 'main' into ch1-issue23

chuawjk committed:
Merge branch 'main' into ch1-issue21

snjannah committed:
Initial draft

chuawjk committed:
Added section on data risk and uncertainties

snjannah committed:
Revise formatting of page

snjannah committed:
Final draft

snjannah committed:
Final draft

chuawjk committed:
Changeed header level

chuawjk committed:
Merge branch 'ch8-issue20' into develop

chuawjk committed:
Merge remote-tracking branch 'origin/ch3-issue19' into develop

chuawjk committed:
Merge remote-tracking branch 'origin/ch7-issue15' into develop

chuawjk committed:
Added to TOC

Commit:
Added #16 + extra requirements to render mermaid.js

Commit:
Changed a phrase to remove skunked word

Commit:
Added #18 + extra requirements to render mermaid.js

Commit:
'with regard to' is the preferred form. Deleted 'very' as what proselint prefers.

Commit:
proselint thinks that 'with regard to' is commercialese, so changing the phrase to something else

chuawjk committed:
Merge remote-tracking branch 'origin/ch5-issue18' into develop

chuawjk committed:
Merge remote-tracking branch 'origin/ch7-issue16' into develop

chuawjk commented:
Thanks for this @Syakyr. I think this is a really important contribution as it gives a comprehensive outline of the technical components of the an AI project.

1. There's a lot of useful content covered here. In keeping with the short-and-sweet intention of the handbook, I'd like to suggestion splitting this into two sections based your basic and advanced workflow diagrams. This way, the sections can separately target new and experienced engineers, respectively.

2. For version control, I was wondering if you'd like to introduce the concept of data versioning (that being said, I understand that there are word count constraints)?

3. Also under version control, I'd suggest omitting the following until we have a standalone article on version control.
```
Refer to another part of this handbook for more information regarding this process.
```

4. The subsection on developer workspace might also benefit from an explicit definition of what a developer workspace is, and what it is used for. The term is quite broad, so I think it will be useful to align readers on exactly what is meant in this article.

For your consideration!

chuawjk committed:
Merge branch 'main' into develop

Commit:
Added #30

chuawjk committed:
Merge remote-tracking branch 'origin/ch5-issue30' into develop

chuawjk committed:
Merge remote-tracking branch 'origin/main' into develop

Commit:
Added .swp files into gitignore

Commit:
Changed header + added reference in 'Start Small' subsection

Commit:
Removed extra 'the'

Commit:
Added contributor(s)

Commit:
Added contributor(s)

snjannah committed:
Add section headers and references

chuawjk committed:
Merge branch 'ch5-issue30' into develop

chuawjk committed:
Merge remote-tracking branch 'origin/ch7-issue16' into develop

chuawjk committed:
Merge pull request #37 from aisingapore/ch3-issue19

Ch3 issue19

chuawjk committed:
Added testing text

chuawjk committed:
More testing text

chuawjk committed:
Merge remote-tracking branch 'origin' into develop

chuawjk committed:
More testing text

Commit:
Merge branch 'test-automerge' into develop

chuawjk committed:
Revert "Test automerge pipeline"

This reverts commit cbb0e1068673022f3f785ace09dc38bbff377f42.

chuawjk committed:
Merge branch 'main' into develop

chuawjk committed:
Test automerge

chuawjk committed:
Merge remote-tracking branch 'origin/develop' into test-automerge

chuawjk committed:
Add test chapter

Commit:
Merge remote-tracking branch 'origin/develop' into automerge-to-develop

chuawjk committed:
Merge branch 'develop' into test-automerge2

chuawjk committed:
Added test chapter 2

chuawjk committed:
Housekeeping

Commit:
Merge remote-tracking branch 'origin/develop' into housekeeping

Commit:
Merge remote-tracking branch 'origin/develop' into housekeeping

chuawjk committed:
Revert "Revert "Test automerge pipeline""

This reverts commit 4cbce96c8187fcc1c3b860abd3151636eb35b82f.

chuawjk committed:
Merge branch 'develop' of https://github.com/aisingapore/ai-practitioner-handbook into develop

Commit:
Merge remote-tracking branch 'origin/develop' into automerge-to-develop

chuawjk committed:
Update test chapter

Commit:
Merge remote-tracking branch 'origin/develop' into test-automerge3

chuawjk committed:
Test automerge

Commit:
Merge remote-tracking branch 'origin/develop' into test-automerge4

chuawjk committed:
Added mermaid.js

Commit:
Merge remote-tracking branch 'origin/develop' into test-automerge4

chuawjk committed:
Addded test ch

Commit:
Merge remote-tracking branch 'origin/develop' into test-automerge5

chuawjk committed:
Addded test ch

Commit:
Merge remote-tracking branch 'origin/develop' into test-automerge6

chuawjk committed:
Added test chapter

Commit:
Merge remote-tracking branch 'origin/develop' into test-automerge7

chuawjk committed:
Added test chapter

Commit:
Merge remote-tracking branch 'origin/develop' into test-automerge8

chuawjk committed:
Added test ch to toc

Commit:
Merge remote-tracking branch 'origin/develop' into test-automerge8

chuawjk committed:
Amended toc

Commit:
Merge remote-tracking branch 'origin/develop' into test-automerge8

chuawjk committed:
Housekeeping

chuawjk committed:
Merge branch 'main' into develop

Commit:
Added link to CI/CD basics for reader's reference

Commit:
Merge remote-tracking branch 'origin/develop' into ch7-issue16

Commit:
Updated overview for chapter 7

Commit:
Merge remote-tracking branch 'origin/develop' into ch7-issue16

Commit:
Split the e2e workflow section into 2, added extra info on the advanced section

Commit:
Merge remote-tracking branch 'origin/develop' into ch5-issue18

Syakyr commented:
> I'd like to suggestion splitting this into two sections based your basic and advanced workflow diagrams. This way, the sections can separately target new and experienced engineers, respectively.

Done 👍🏼 

> For version control, I was wondering if you'd like to introduce the concept of data versioning (that being said, I understand that there are word count constraints)?

I've added a small part within the version control section for data versioning, and refer to another website for now, unless we have a standalone article on data versioning.

> I'd suggest omitting "Refer to another part of this handbook for more information regarding this process." until we have a standalone article on version control.

Done 👍🏼 

> 4. The subsection on developer workspace might also benefit from an explicit definition of what a developer workspace is, and what it is used for. The term is quite broad, so I think it will be useful to align readers on exactly what is meant in this article.

Added a small part with regards to this section to point to the existence of remote developer workspaces.

> Could I trouble you to rephrase the section title as a question?

Done 👍🏼 

> I was wondering if this section might be a better fit for another section. The current chapter was intended to cover DataOps and EDA process, but this section provides a broader view. My suggestions is to move it to Chapter 3 Collaborative Development Platforms, but I'm aware that's not a perfect fit either. What do you think?

This could be moved to Chapter 2 on Project Management & Technical Leadership instead, though not sure what user story this would fall under.

Commit:
Revamped topic following discussion

Commit:
Merge remote-tracking branch 'origin/develop' into ch5-issue30

Commit:
Reworded to meet proselint requirements

Commit:
Merge remote-tracking branch 'origin/develop' into ch5-issue30

ngkh2018 commented:
Basic Workflow

1. A hyperlink seems to be missing in this sentence (‘You can check this section for more information on data storage options.’) under the ‘Data Storage’ section.

2. There are external links embedded in the content. Do we want to include them under the ‘References’ section as well? Will this be seen as repeated links? Same comment for ‘Enhanced Workflow’.

3. Would including links (under the ‘References’ section) to REST and gRPC be useful to the readers?

Enhanced Workflow

4. It seems like we did not cover ‘Experiment & Pipeline Tracking’. Do we want to?

Commit:
Updated e2e-workflow sections

Syakyr commented:
> 1. A hyperlink seems to be missing in this sentence (‘You can check this section for more information on data storage options.’) under the ‘Data Storage’ section.

Fixed. It's for linking to #30, but I forgot to update that. 

> 2. There are external links embedded in the content. Do we want to include them under the ‘References’ section as well? Will this be seen as repeated links? Same comment for ‘Enhanced Workflow’.

I've repeated links with #30, so I'll do that here as well. Markdown supports reference linking, so it's not much of a hassle to change, or rather, it looks cleaner in code.

> 3. Would including links (under the ‘References’ section) to REST and gRPC be useful to the readers?

Because the handbook is meant to be agnostic in its solutions, I've only put REST and gRPC protocols as examples, of which I would assume that the reader would go read further on what those are, and discover what wrappers are available to set up those protocols with their programming language of choice. We could improve on this, but I would prefer to get some feedback that they would need more guidance with regards to how those protocols work, with Python as an example, for instance, or how those protocols interact with other programs or the user I suppose.

> 4. It seems like we did not cover ‘Experiment & Pipeline Tracking’. Do we want to?

Seems that I've missed that section while splitting the articles into two. I've added that together with the Model Monitoring section since it's quite similar while targeting different processes.

Commit:
Fixed header issues for e2e-workflow-adv

kevinosl commented on book/5-data-mgmt-exp-proc/e2e-workflow.md:
```diff
@@ -0,0 +1,155 @@
+
+For AI practitioners such as ourselves, there is also a subset of 
+version control that targets toward datasets, which you may read 
+[this article][data-versioning] for more information.
```
```which you may read 
[this article][data-versioning] for more information.```

kevinosl replied:
I think the link to the article still appears as a placeholder. May wish to check.

Syakyr replied:
I've fixed this issue; it seems that there's a typo on line 155 that breaks the hyperlinks.

kevinosl commented on book/5-data-mgmt-exp-proc/e2e-workflow.md:
```diff
@@ -0,0 +1,155 @@
+# What are the Processes Involved to Build a Basic End-to-End Workflow?
```
Would this sound better?
# What are the Processes Involved in Building a Basic End-to-End Workflow?

Syakyr replied:
Yup. I've changed the title to the recommended one 👍🏼

kevinosl commented:
General comment: were the diagram(s) created ourselves, or taken from a framework? If its the latter, it would be good to have a caption to reference the framework and also place the link in the References.
If its the former, it might also still be good to add some references to industry-accepted workflow frameworks.
(PS: by workflow frameworks I am referring not to TDSP or CRISP-DM, which personally I consider to be more of high-level lifecycle frameworks rather than workflow frameworks. Those may sit well in the PM chapter instead. Not sure if you guys have a different view.)

Syakyr committed:
Fixed typo in line 155

Syakyr committed:
Changed section phrase to flow better

Syakyr committed:
Colour coded the notes for the advanced e2e section

Syakyr commented:
> General comment: were the diagram(s) created ourselves, or taken from a framework? If its the latter, it would be good to have a caption to reference the framework and also place the link in the References.
> 
> If its the former, it might also still be good to add some references to industry-accepted workflow frameworks. (PS: by workflow frameworks I am referring not to TDSP or CRISP-DM, which personally I consider to be more of high-level lifecycle frameworks rather than workflow frameworks. Those may sit well in the PM chapter instead. Not sure if you guys have a different view.)

The diagram was created from Ryzal's slides with regards to our MLOps framework, but with the product names removed to ensure generality within the handbook. I've linked to the ml-ops.org website should readers want to know more with regards to end-to-end workflows and MLOps in general, although it's not just pertaining to the end-to-end workflow. I can give a direct link to the section on the workflows, but I would assume that they've read both sections and want to know more, and they'll go to the website linked at the end of the advanced workflows section, so it might be redundant to add references on the 1st section.

> Lastly, it might be good to color code the basic and enhanced nodes in the combined diagram, so readers can orientate themselves more easily at one glance. This may be especially helpful in the Enhanced article, which only discusses the nodes that are not captured in the Basic article.

I've used a different color shade to denote the advanced nodes in both sections to differentiate between the basic and enhanced nodes 👍🏼

ryzalk commented:
Closing this pull request in view of the new merging workflow i.e. from contributing branches to `develop`.
CC @chuawjk
